.github/workflows/ci.yml

```diff
@@ -0,0 +1,46 @@
+name: CI
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    
+    strategy:
+      matrix:
+        node-version: [18.x, 20.x, 22.x]
+    
+    steps:
+    - uses: actions/checkout@v4
+    
+    - uses: actions/setup-node@v4
+      with:
+        node-version: ${{ matrix.node-version }}
+    
+    - name: Install dependencies
+      uses: wyvox/action-setup-pnpm@v3
+      with:
+        pnpm-version: 10.11.0
```

Comment:
refactor this out into a separate action like this file here: https://github.com/rip-technologies/shipstation-node/commit/101b4e29f8b7a7344e1447adcbfd69a253f238e8#diff-342d59190b4737ee45e2062eb625ada477bcea5b4a843b25900ad55d7982f200